### ISSUE TITLE
Exception from Deps afterFlush function function: TypeError: undefined is not a function

### ISSUE THREAD

**Issue by rmarmer1** [frozeman/meteor-animation-helper-velocity#2]

**Comment by rmarmer1** [comment]
Can't tell you how thrilled I was to run across your latest creations! Using a simple three page (template) app with Iron-router & 0.8.3-rc5, I had no problem getting your new meteor-animation-helper to work. 

But when I set up the same basic structure & added meteor-animation-helper-velocity along with the suggested markup,

```
<div data-animate data-properties="left, opacity" data-from-values="1000px, 0" data-to-values="0px, 1">
    ...
</div>
```

inside the tags

{{#AnimateWithVelocity}}

{{/AnimateWithVelocity}}

I got the "Exception from Deps afterFlush function" error which then increments whenever I click a link. I've included a screenshot...

![image](https://cloud.githubusercontent.com/assets/4955633/3760136/93f4af72-1870-11e4-8034-14f931c55531.png)

My nav links are set up as;

```
     <a href="{{pathFor 'one'}}"> One</a>  
```

and the buttons use; 

```
Template.goThree.events({
  'click #clickme': function () {
    Router.go('/three');
  } 
```

Could my use of iron-router be causing/contributing to the problem?

THANKS!

**Comment by frozeman** [comment]
Could you please send me you simple app at fabian@frozeman.de, or click on the first line of the exception to see where in the Animation..Velicity.js the error was generated?

**Comment by rmarmer1** [comment]
Sure. I'ts rather late here, I'll send you the app tomorrow... For now, here is the code from the 1st line of the error:

![image](https://cloud.githubusercontent.com/assets/4955633/3761800/02b9aa48-188f-11e4-90c7-b53f3fcdc001.png)

The error was on line 179.
Thanks again!

**Comment by frozeman** [comment]
I updated the package, can you please update and check if the error still occurs?

The `underscore-string-latest` package was missing. In my projects i add them manually so i didn't encounter this error.

**Comment by rmarmer1** [comment]
Indeed! My simple app, with the provided markup, is working beautifully!

Next, I'll try using Velocity more fully, but based on what I see now, I anticipate smooth sailing with animation-helper-velocity from here on out. I also plan to experiment with integrating animate.css into your meteor-animation-helper package.

Pre & post Meteor 0.8.0, there have been any number of straight-ahead ways to implement "animate-in" route-changes via iron-router. But the introduction of Blaze seemed to make the "animate-out/animate-in" combination exceedingly painful & pretty much impossible without some very ugly hacks.  

At first I had no idea what I was up against. I've since gained some insight into the situation & am under the impression that a proper fix (to Blaze and/or Iron-Router) will be forthcoming by, or perhaps soon after Meteor's 1.0 release.

That said, I know I'm not alone in having spent many fruitless & frustrating hours in experimentaion.

So, in short @frozeman, you're a prince! 

I've not been to Europe in many years, but should the opportunity present itself, I'd certainly appreciate the chance to buy you the traditional "open-source beer"!

Thanks so much again for all you do!

Randy Marmer

**Comment by frozeman** [comment]
Thanks for the cud's :)
If you come to berlin let me know.

I'm also very happy with the hooks now, as before it was a pain the animation. I'll tied to make it possible withe `template-animation-helper` package, but it was a strong hack and not so pretty, now with hooks and velocity it works like a charm.
